src/middlewared/middlewared/plugins/network.py

```diff
@@ -1292,14 +1292,15 @@ async def do_delete(self, oid):
         await self.middleware.call('network.common.check_failover_disabled', schema, verrors)
 
         if iface := await self.get_instance(oid):
-            if iface['type'] == 'LINK_AGGREGATION':
-                filters = [('type', '=', 'VLAN'), ('vlan_parent_interface', '=', iface['id'])]
-                if vlans := ', '.join([i['name'] for i in await self.middleware.call('interface.query', filters)]):
-                    verrors.add(schema, f'The following VLANs depend on this interface: {vlans}')
-            if iface['type'] == 'BRIDGE' and (
+            filters = [('type', '=', 'VLAN'), ('vlan_parent_interface', '=', iface['id'])]
+            if vlans := ', '.join([i['name'] for i in await self.middleware.call('interface.query', filters)]):
+                verrors.add(schema, f'The following VLANs depend on this interface: {vlans}')
+            elif iface['type'] == 'BRIDGE' and (
                 iface['name'] == (await self.middleware.call('virt.global.config'))['bridge']
             ):
                 verrors.add(schema, 'Virt is using this interface as its bridge interface.')
+        else:
+            verrors.add(schema, f'{oid!r} does not exist')
```

Comment:
`get_instance` throws an exception (`InstanceNotFound`) when the instance is not found, it does not return `None`